### PR TITLE
ENHANCEMENT - Dockerized Server and DB

### PR DESCRIPTION
**Pull request by Soldatstar**
This pull request addresses issue #3.

### Changes Made
- Added a Dockerfile that defines a build stage to package into the .war file
- Included a second deploy stage in the Dockerfile to serve the application via a Tomcat server on port 8080 with the path `/api`
- Added the docker-compose.yml file to include a MySQL database and environment variables for the HTTP server
- Added a new environment variable MYSQL_ROOT_PASSWORD for the MySQL database
### Instructions
To start the services, execute the following command:
```bash
docker compose -f docker-compose.yml -p opentoonix-http-server up -d
```

Please review and merge as appropriate. Thank you!

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  db:
+    image: mysql:8.3.0
+    restart: always
+    environment:
+      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
+      MYSQL_DATABASE: ${DATABASE_NAME}
+      MYSQL_USER: ${DATABASE_USERNAME}
+      MYSQL_PASSWORD: ${DATABASE_PASSWORD}
+    volumes:
+      - db_data:/var/lib/mysql
```
Comment (Juansecu) [inline docker-compose.yml:14]: Consider the possibility of including an environment variable to specify the local path where the database data will be saved in the host machine.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    environment:
+      AVATAR_STORAGE_FOLDER_PATH: ./avatars
+      DATABASE_HOST: db
+      DATABASE_NAME: ${DATABASE_NAME}
+      DATABASE_PASSWORD: ${DATABASE_PASSWORD}
+      DATABASE_PORT: 3306
+      DATABASE_USERNAME: ${DATABASE_USERNAME}
+      JWT_SECRET: ${JWT_SECRET}
+    volumes:
+      - .:/app
```
Comment (Juansecu) [inline docker-compose.yml:33]: Consider the possibility of include individual volumes to each part of the application that needs them. (e.g. A volume for avatars, and another one for static assets).

**Review by Juansecu — CHANGES_REQUESTED** [review]
Please apply the requested changes, so we can accept them.

**Comment by Soldatstar** [comment]
I've added DB_DATA_PATH for the host volume and changed to AVATAR_STORAGE_FOLDER_PATH for the Avatar volume. Also, included a volume for static assets and a variable STATIC_FILES_PATH if necessary

**Comment by Juansecu** [comment]
@Soldatstar Thank you, and you can remove the volume for the static assets. It was just an example, so you could get my point.

**Comment by Soldatstar** [comment]
Done!

**Comment by Juansecu** [comment]
@Soldatstar Sorry, I was testing your changes, and unfortunately, Docker Compose does not accept environment variables for volumes. Could you specify the module as you did before, please?

By the way, I wanted to ask you... What is the purpose of executing the command `mvn dependency:go-offline -B` when building the application in the Dockerfile?

**Comment by Soldatstar** [comment]
@Juansecu 
The command mvn dependency:go-offline -B resolves all dependencies and is supposed to aid in caching them, making rebuilding faster, but I haven't tested it yet.

Regarding the Docker Compose issue, it's strange. I had placed mine in an .env file, and Docker was able to read everything, including the Avatar path. Maybe an format issue?

**Comment by Juansecu** [comment]
I mean, the environment variable works inside the service, but volumes must be specified inside the `volumes` attribute, too. This is where Docker does not accept environment variables. So could you specify static path to local volumes , please?

**Comment by Juansecu** [comment]
@Soldatstar I mean, the environment variable works inside the service, but volumes must be specified inside the volumes attribute, too. This is where Docker does not accept environment variables. So could you specify static path to local volumes , please?

**Comment by Soldatstar** [comment]
I put the path from the readme as the static path for DB and Avatars

` /c/Users/usr/OpenToonix`

is it fine like this?

**Comment by Juansecu** [comment]
@Soldatstar Nop. Leave them as `mysql` for the database volume, and `avatars` for the avatars volume. By the way, could you change `db` service name to `mysql` and change the tag of the MySQL image to `8`, so it always download the latest 8th version of MySQL, please?

By the way, you may have to consider of remove the **Tomcat** stuff from the Dockerfile, since Spring Boot already includes Tomcat, so it is possible to run the application directly with a JRE image.

**Comment by Juansecu** [comment]
@Soldatstar Please revert the change of the pom.xml to avoid problems with the functionality, since it actually depends on a Servlet. I will do that change when I update the entire application to more recent versions of Java and Spring Boot.

By the way, check out the commit standards. You are not following them correctly.

**Comment by Soldatstar** [comment]
Thanks for the heads-up, I haven't worked with .wars before, wasn't aware that its also executable with JRE

**Comment by Juansecu** [comment]
@Soldatstar Np! And thanks for your contribution!

**Pull request merged**